app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/adapters/DashboardCardsAdapter.java

```diff
@@ -0,0 +1,153 @@
+
+        holder.collapsedCard.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+
+                holder.collapsedCard.animate().alpha(0f).start();
+                holder.collapsedCard.setVisibility(View.GONE);
+                holder.expandedCard.setVisibility(View.VISIBLE);
+            }
+        });
+
+        holder.expandedCard.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                holder.collapsedCard.animate().alpha(1f).start();
+                holder.collapsedCard.setVisibility(View.VISIBLE);
+                holder.expandedCard.setVisibility(View.GONE);
+            }
+        });
+    }
+
+    private void setExpandedCardValues(ViewHolder holder, DashboardCardsModel model) {
+        holder.cardNameExpandedText.setText(model.getName());
+
+        holder.budgetedExpandedText.setText(getCalculatedValue(model.getBudgetedAmount()));
+        holder.availableExpandedText.setText(getCalculatedValue(model.getAvailableAmount()));
+        holder.spentExpandedText.setText(getCalculatedValue(model.getSpentAmount()));
+
+        if(holder.categoriesDynamicLayout.getChildCount() > 0){
+            holder.categoriesDynamicLayout.removeAllViews();
+        }
+
+        for(int i=0; i<model.getCategoryName().size(); i++){
+            View view = LayoutInflater.from(context).inflate(R.layout.expanded_categories_data_card, null);
+
+            TextView categoryName, budgetedAmount, availableAmount, spentAmount;
+
+            categoryName = view.findViewById(R.id.categoryName_expandedCategory);
+            budgetedAmount = view.findViewById(R.id.budgetedBalanceText_expandedCard);
+            availableAmount = view.findViewById(R.id.availableBalanceText_expandedCard);
+            spentAmount = view.findViewById(R.id.spentBalanceText_expandedCard);
+
+            categoryName.setText(model.getCategoryName().get(i));
+            budgetedAmount.setText(model.getBudgetedAmount().get(i));
+            availableAmount.setText(model.getAvailableAmount().get(i));
+            spentAmount.setText(model.getSpentAmount().get(i));
+
+            holder.categoriesDynamicLayout.addView(view);
+        }
+
+    }
+
+    private void setCollapsedCardValues(ViewHolder holder, DashboardCardsModel model) {
+        holder.cardNameCollapsedText.setText(model.getName());
+
+        holder.availableCollapsedText.setText(getCalculatedValue(model.getAvailableAmount()));
+        holder.spentCollapsedText.setText(getCalculatedValue(model.getSpentAmount()));
+
+        holder.categoriesAvailableCollapsed.setText(String.format("View %d categories", model.getCategoryName().size()));
+    }
+
+    private String getCalculatedValue(List<String> amountList) {
+        double value = 0.00;
+        for (int i = 0; i < amountList.size(); i++) {
+            value += Float.parseFloat(amountList.get(i).replace("$", ""));
```

Comment:
There are users from around the world using the app. You cannot assume the currency will be always be $. You have to check the users locale and set the currency automatically. Message me on Slack to discuss this issue.

Reply:
Okay